Search.py

```diff
@@ -236,6 +236,64 @@ def can_beat_game(self, scan_for_items=True):
         else:
             return False
 
+    def can_beat_goals(self, category_name, goals, scan_for_items=True):
+        valid_goals = []
+        test_goals = list(goals)
+        for goal in test_goals:
+            beaten = True
+            
+            # Check if already beaten
+            if (goal.items is not None):
+                for i in goal.items:
+                    # Every world has the same set of goals, but reachable item quantity varies per-world in beatable only
+                    # Avert your eyes, here be spaghetti
+                    beaten = (beaten and all(map(State.has, self.state_list, itertools.repeat(i['name']), [state.world.goal_categories[category_name].get_goal(goal.name).get_item(i['name'])['quantity'] for state in self.state_list])))
+            if (goal.locations is not None):
+                for l in goal.locations:
+                    beaten = (beaten and all(map(self.can_reach_spot, self.state_list, itertools.repeat(l['name']), itertools.repeat(l['age']))))
```

Comment:
Much as I love both pasta and calling the `map` function without lambdas, it's probably better here:
```suggestion
                    beaten = beaten and all(map(
                        lambda s: s.has(
                            i['name'],
                            s.world.goal_categories[category_name].get_goal(goal.name).get_item(i['name'])['quantity']),
                        self.state_list))
            if goal.locations:
                for l in goal.locations:
                    beaten = beaten and all(map(
                        lambda s: s.can_reach_spot(l['name'], l['age']),
                        self.state_list))
```

(And honestly, maybe `s.world.goal_categories[category_name].get_goal(goal.name).get_item(i['name'])['quantity'])` should be a helper function in `State`.)